src/openforms/submissions/tests/form_logic/test_submission_logic.py

```diff
@@ -773,6 +773,11 @@ def test_evaluate_logic_with_default_values(self):
         )
 
     def test_evaluate_logic_log_event_triggered(self):
+        # Logs need to be cleared, otherwise an AssertionError is raised in the
+        # logs.count test due to logevents created elsewhere for plugins with
+        # empty prefill values
+        TimelineLogProxy.objects.all().delete()
```

Comment:
I suspect this is because of the threadpool that's used in the prefill functionality. Therefore, (prefill) tests should instead add a cleanup function in the `setUp`:

```py
def setUp(self):
    super().setUp()

    self.addCleanup(lambda: TimelineLogProxy.objects.all().delete())
```

Reply:
I've tried several ways to add the cleanup method to the testcases that deal with prefill plugins so as to treat the source of the failing tests rather than the symptoms, but to no avail. The only thing that works for me is clearing the logs in the functions that fail.